bach/bach/dataframe.py

```diff
@@ -2439,6 +2434,68 @@ def drop_duplicates(
         self._update_self_from_df(df)
         return None
 
+    def dropna(
+        self,
+        axis: int = 0,
+        how: str = 'any',
+        thresh: Optional[int] = None,
+        subset: Optional[Union[str, Sequence[str]]] = None,
+        inplace: bool = False
+    ) -> Optional['DataFrame']:
+        """
+        Removes rows with missing values.
+
+        :param axis: only ``axis=0`` is supported. This means rows that contain missing values are dropped.
+        :param how: determines when a row is removed. Supported values:
+           - 'any': rows with at least one missing value are removed
+           - 'all': rows with all missing values are removed
+        :param thresh: determines the least amount of missing values a row needs to be removed
+        :param subset: series label or sequence of labels to be considered for missing values
+        :param inplace: Perform operation on self if ``inplace=True``, or create a copy.
+
+        :return: a new dataframe with dropped rows if inplace = False, otherwise None.
+        """
+        if axis:
+            raise ValueError('only axis = 0 is supported.')
+
+        if how not in ('any', 'all'):
+            raise ValueError(f'{how} is not a valid value for "how" parameter.')
+
+        subset = self._get_parsed_subset_of_data_columns(subset)
+        dropna_series = subset or self.data_columns
```

Comment:
This seems more in line with the pandas behaviour
```suggestion
        dropna_series = self.data_columns if subset is None else subset
```

Reply:
`if subset = []` then pandas checks 0 columns, we check all columns.

```python
pdf = pd.DataFrame(
    data={
        'a': [1, 2, 3, 4, 5, 6, np.nan, 7., 0, 0],
        'b': [11.9, 32.0, 43.123, 64.425124, 25.00000000001, 6, np.nan, 78., 0, 0],
        'c': [0, None, 10, 11, 11, 11, 11, 11, 11, 15],
        'd': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j'],
        'e': [True, True, True, True, True, True, True, True, True, False],
    },
)
df = DataFrame.from_pandas(engine=engine, df=pdf.reset_index(drop=False), convert_objects=True, materialization='table', if_exists='replace', name='name of the test table')
df = df.reset_index(drop=True)
df = df.drop(columns=['index'])
print(len(pdf.dropna(subset=[])))  # 10
print(len(df.dropna(subset=[]).to_pandas()))  # 8
```

Reply:
I think raising an exception is a worse case: then our end user has to handle a special case. 

I can imagine that if an end users writes some function that iterates over some values for a parameter, then they don't want to handle special cases where that parameter is an empty list; much nicer if the behaviour stays consistent in that case.

Reply:
well, in that case we should just return a copy of a df? I find this behavior a bit weird/non-intuitive imho, guess that we need to specify that in the docs

Reply:
I guess just a copy would be good. Let's discuss tomorrow in person if you find this non-intuitive, as I feel it's the other way around :)